### ISSUE TITLE
piechart - slice at 0% take 100% of the piechart

### ISSUE THREAD

**Issue by oliverlj** [amcharts/amcharts5#192]
wrong chart : 
![image](https://user-images.githubusercontent.com/1826498/148518128-bb394454-965a-4a9a-bd5c-b8f18e3828ef.png)

Expected :
![image](https://user-images.githubusercontent.com/1826498/148518235-79f8ad26-7c00-4da8-be9f-685d4d0c9516.png)

for the moment, I'm hiding small slice with this :
```ts
this.series.slices.template?.adapters.add('visible', (_value, target) => {
      if (target) {
        const dataItem = (target as unknown as RadialLabel).dataItem;

        if (dataItem) {
          const settings = dataItem['_settings'] as IPercentSeriesDataItem;

          return settings.valuePercentTotal > 0.01;
        }
      }

      return false;
    });
```

whole chart :
```
let chart = root.container.children.push(
      PieChart.new(root, {
        startAngle: 180,
        endAngle: 360,
        layout: root.verticalLayout,
        innerRadius: percent(50),
        radius: percent(60),
      }),
    );

    // Create series
    // https://www.amcharts.com/docs/v5/charts/percent-charts/pie-chart/#Series
    // start and end angle must be set both for chart and series
    this.series = chart.series.push(
      PieSeries.new(root, {
        startAngle: 180,
        endAngle: 360,
        valueField: this.args.valueField,
        categoryField: this.args.categoryField,
        alignLabels: false,
      }),
    );

    this.series.states.create('hidden', {
      startAngle: 180,
      endAngle: 180,
    });

    this.series.slices.template.setAll({
      cornerRadius: 5,
    });

    this.series.ticks.template.setAll({
      forceHidden: true,
    });

    this.series.labels.template.adapters.add('visible', (_value, target) => {
      if (target) {
        const dataItem = (target as unknown as RadialLabel).dataItem;

        if (dataItem) {
          const settings = dataItem['_settings'] as IPercentSeriesDataItem;

          return settings.valuePercentTotal > 5;
        }
      }

      return false;
    });

    this.series.slices.template?.adapters.add('visible', (_value, target) => {
      if (target) {
        const dataItem = (target as unknown as RadialLabel).dataItem;

        if (dataItem) {
          const settings = dataItem['_settings'] as IPercentSeriesDataItem;

          return settings.valuePercentTotal > 0.01;
        }
      }

      return false;
    });

this.series.data.setAll(this.args.data?.filter((slice) => slice[this.args.valueField] > 0));
      this.series.appear(1000, 100);
```

**Comment by zeroin** [comment]
Could you create a codepen with the chart + data?

**Comment by oliverlj** [comment]
https://codepen.io/oliverlj/pen/VwMBBya

**Comment by martynasma** [comment]
Fixed in [5.0.21](https://github.com/amcharts/amcharts5/releases/tag/5.0.21).

## [5.0.21] - 2022-01-11

### Added
- New `Sprite` setting: `forceInactive`. If set to `true`, it will override `interactive` setting as well as ignore any interaction events set on the element.
- `cursorhidden` event added to `XYCursor`. Dispatched when user rolls-out of the chart plot area and cursor is hidden.
- New `Component` method: `markDirtyValues()`. Forces a repaint of the element which relies on data.

### Changed
- Value (4th parameter of the Heat rule's `customFunction`) will now receive actual value of the target data item, not interpolated calculated setting value.

### Fixed
- Fixing bug when not passing the duration to the `animate` method ([Issue 181](https://github.com/amcharts/amcharts5/issues/181)).
- Horizontal and vertical layouts with `width`/`height` set in percents plus `minWidth`/`minHeight`/`maxWidth`/`maxHeight` set were arranging items incorrectly.
- Very small slices with `cornrerRadius > 0` were being drawn as a full circle.
- `SliceGrouper` with was not zooming out properly if there was no legend in chart.
- changing `groupData` from `false` to `true` on DateAxis was not working when doing so after the chart was built.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
